### DISCUSSION TITLE
Is there a example about how to deal with Color in Rules

### DISCUSSION THREAD

**Discussion by s3xysteak** [unocss/unocss#3520] [Q&A]
Like a rule `/^highlight-(.*)$/ `, the captured part expects a color such as `highlight-white/50` or `highlight-#fff/50`. I can get the color value with args[0][1], but how can I extract the color from these rules? Is there a tool or function available to handle this?
Maybe like
```javascript
([_, value], { colorResolver }) => {
  const color = colorResolver(value)
  return {
    boxShadow: `inset 0 1px 0 0 ${color}`
  }
}
```

**Comment by chu121su12** [comment]
You may want to check out the preset implementation
https://github.com/unocss/unocss/blob/96aba8735118360aafb22018d2b50188956b6a00/packages/preset-mini/src/_rules/typography.ts#L17
